File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution(object):
    def partitionString(self, s):
        """
        :type s: str
        :rtype: int
        """
        curr = ""
        totalcount = 0
        while(len(s) !=0):
            if(s[0] in curr):
                curr = ""
                totalcount += 1
                curr += s[0]
            else:
                curr += s[0]
            s = s[1:]
        if(len(curr) != 0):
            totalcount += 1
        return totalcount

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        int n = s.length();
        std::unordered_set<char> seen;
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (seen.count(s[i])) {
                seen.clear();
                count++;
            }
            seen.insert(s[i]);
        }
        return count + 1;

    }
};